src/proxyConnect.ts

```diff
@@ -0,0 +1,132 @@
+import { Socket } from 'net'
+import { DOMParser } from 'xmldom'
+import { EventEmitter } from 'events'
+import { decode } from 'iconv-lite'
+import { ENCODING } from './xdebugConnection'
+
+export const DEFAULTIDEKEY = 'vsc'
+export interface ProxyMessages {
+    defaultError: string
+    deregisterInfo: string
+    deregisterSuccess: string
+    duplicateKey: string
+    nonexistentKey: string
+    registerInfo: string
+    registerSuccess: string
+    resolve: string
+    timeout: string
+}
+
+/** Informs proxy of incoming connection and who to pass data back to. */
+export class ProxyConnect extends EventEmitter {
```

Comment:
This class has to do very similar things as `XDebugConnection` (manage a socket, parse XML responses, expose methods to send commands), yet doesn't use reuse any of the logic from `DBGPConnection`. This exposes a couple of problems that `DBGPConnection` handles, but `ProxyConnect` does not. Listing out in comments below. The proper solution would likely be that this class should just `extend DBGPConnection` and reuse its logic.

Reply:
Although that would be ideal, the reality is, the proxy ide protocol is NOT the same as the dbgp protocol. The specification is loos here and the first major implementation went its own way and now we have a defacto standard. See this issue, where I ask the original author to clarify  this in the specs: https://github.com/derickr/dbgp/issues/19

I will however try to make the class behave more like `XDebugConnection`.